### PR TITLE
Initial project structure

### PR DESCRIPTION
**Pull request by monikasudol**
Hi Guys!

There is a initial project structure, add style will be required. Please look at this in your free time.

### CODE REVIEW

File: src/modules/registration/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        <InputWrapper text="Confirm password"
+          type="password"
+          sendDetail={this.handlePasswordConfimation}
+          placeholder="*******" />
+        <InputWrapper text="Email"
+          type="mail"
+          sendDetail={this.handleMail}
+          placeholder="mail@example.com" />
+        <button className='registartion-button-resume'
+          onClick={(username, password, mail) =>
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/modules/registration/index.js]: right, i've improved that

File: src/modules/registration/index.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  };
+
+  handleMail = (mail) => {
+    this.setState({
+      mail
+    });
+  };
+
+  handlePasswordConfimation = (password) => {
+    {
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/modules/registration/index.js]: i've changed this function

**Comment by monikasudol** [comment]
Thanks a lot for CR, I'm pushing changes and there are some other small improvements

File: src/App.js [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+import { connect } from 'react-redux';
 import WelcomePage from './modules/welcome-page';
 import Main from './modules/main';
 import './App.css';
 
-class App extends Component {
+class App extends PureComponent {
+
   render() {
+    const { userIsActive } = this.props;
```
Comment (yuriyyakym) [inline src/App.js]: `isUserActive` is a better name

File: src/App.js [hunk 4]
```diff
@@ -1,28 +1,25 @@
...
-            <Route exact path='' component={WelcomePage} />
-          </Switch>
-        </BrowserRouter>
-        <BrowserRouter>
-          {
-            userIsActive &&
-            (<Redirect to="/me" />)
-          }
-        </BrowserRouter>
+        <Switch>
```
Comment (yuriyyakym) [inline src/App.js]: You do not need switch for a single route

File: src/App.js [hunk 5]
```diff
@@ -1,28 +1,25 @@
...
+          <Route path='/me' component={Main} />
+        </Switch>
+        <WelcomePage />
+        {
+          userIsActive &&
```
Comment (yuriyyakym) [inline src/App.js]: Improve formatting

File: src/modules/welcome-page/index.js [hunk 6]
```diff
@@ -10,15 +11,13 @@ class WelcomePage extends Component {
     return (
       <div className='welcome-page-container'>
         <Header />
-        <BrowserRouter>
-          <Switch>
-            <Route path='/registration' component={Registration} />
-            <Route path='/login' component={Login} />
-          </Switch>
-        </BrowserRouter>
+        <Switch>
```
Comment (yuriyyakym) [inline src/modules/welcome-page/index.js]: We do not need switch in this case. There is no possibility that two routes will match.

File: src/api/registration.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { post, toJSON } from '../utils/index.js';
+
+export const registerUser = (name, username, surname,
```
Comment (yuriyyakym) [inline src/api/registration.js]: very strange formatting :)

File: src/modules/registration/index.js [hunk 8]
```diff
@@ -8,63 +8,94 @@ class Registration extends PureComponent {
...
+    this.setState({ mail });
+  };
+
+  handleConfirmedPassword = (confirmedPassword) => {
+    this.setState({ confirmedPassword });
   };
 
-  handlePasswordConfimation = (password) => {
+  onAddingNewUser = () => {
+    const { name, surname, username, password,
```
Comment (yuriyyakym) [inline src/modules/registration/index.js]: strange formatting

File: src/modules/registration/index.js [hunk 9]
```diff
@@ -8,63 +8,94 @@ class Registration extends PureComponent {
...
+      confirmedPassword, mail } = this.state;
     {
-      this.state.password === password
-        ? console.log('Good')
-        : console.log('Wrong')
+      (password === confirmedPassword)
```
Comment (yuriyyakym) [inline src/modules/registration/index.js]: It's better not to evaluate functions from ternary. Better approach is to use if statement

File: src/state/current-user/saga.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+import { registerUser } from '../../api/registration';
+
+export default function* () {
+  yield takeEvery(ADD_NEW_USER, onAddingNewUser);
+};
+
+function* onAddingNewUser(action) {
+  const { payload } = action;
+  try {
+    yield call(registerUser, payload.name, payload.surname,
```
Comment (yuriyyakym) [inline src/state/current-user/saga.js]: Why not to pass whole payload to `regiserUser` function?

Comment (yuriyyakym) [reply src/state/current-user/saga.js]: It's not a good practice to have a lot of arguments in a function

**Pull request merged**